### DISCUSSION TITLE
Success! (Arch+Hyprland+Distrobox)

### DISCUSSION THREAD

**Discussion by gestalt73** [cryinkfly/Autodesk-Fusion-360-for-Linux#492] [General]
Wine:  9.19.1 (also confirmed under 9.18)

Man, from the install to capturing the token I could not get this to work under ubuntu or fedora.

But running this under wayland + hyprland is a thing of beauty.  totally usable.  The project browser appears as a separate window, and rightclick instantiates the popup menu behind the browser window, but a simple left click + hold brings it to the front.

I'm glad I installed this under distrobox to keep everything separate.

Here's my fusion.desktop file and icon for anyone else who wants it.

Thanks!

~/.local/share/applications/fusion.desktop

[Desktop Entry]
Name=Fusion
GenericName=Terminal entering Fusion
Comment=Terminal entering Fusion
Categories=Distrobox;System;Utility
Exec=/usr/bin/distrobox enter --name  fusion  --no-tty  --  /home/alansrobotlab/.autodesk_fusion/bin/autodesk_fusion_launcher.sh %F 
Icon=/home/alansrobotlab/.local/share/icons/fusion.png
Keywords=distrobox;
NoDisplay=false
Terminal=false
TryExec=/usr/bin/distrobox
Type=Application
 
 
 
![screenshot_08102024_145347](https://github.com/user-attachments/assets/2f47f75d-edda-40e2-be74-adac8707e895)

![fusion](https://github.com/user-attachments/assets/c49349c6-b25a-4ea0-a13a-0bc6853bb72b)

**Comment by NischalDawadi** [comment]
Is it still running well. Can you help me replicate your setup... I keep running with bad fonts and black bars especially mouse is jittery and cannot create sketch as well.

**Comment by alansrobotlab2** [comment]
The short answer is yes.  :-)

![fusion pepito render screenshot](https://github.com/user-attachments/assets/a3cf6f4d-17ec-4c3f-8af6-7f29f40cdc00)

All the important stuff works, the browser window is quirky etc...
The longer answer is that it's complicated enough that I had to write down the steps.

This is on an arch system with nvidia graphics, btw.  ;-)

**Step 1:** 
```
distrobox create --name fusion --image archlinux:latest

distrobox enter fusion
```

**Step 2:**
```
sudo pacman -S nano

TERM=xterm sudo nano /etc/pacman.conf
```

Append below to pacman.conf
```
[multilib]
Include = /etc/pacman.d/mirrorlist
```


**Step 3:**
```
sudo pacman -Syu

sudo pacman-key --init
sudo pacman-key --populate archlinux

sudo pacman -Syu eza fzf fastfetch base-devel mokutil nvidia

sudo pacman -Syu wine wine-mono wine-gecko

rm -rf .autodesk_fusion

curl -L https://raw.githubusercontent.com/cryinkfly/Autodesk-Fusion-360-for-Linux/main/files/setup/autodesk_fusion_installer_x86-64.sh -o "autodesk_fusion_installer_x86-64.sh" && chmod +x autodesk_fusion_installer_x86-64.sh && ./autodesk_fusion_installer_x86-64.sh --install --default
```

Fusion comes up after a while...

Click on Login to get the login window on the browser...

Then login...

Then right click on the link and grab the URL from the browser...

**Step 4:**
```
distrobox enter fusion

cd .autodesk_fusion/wineprefixes/default/drive_c/Program\ Files/Autodesk/webdeploy/production/WHATEVER THE LONGEST FOLDER NAME IS/Autodesk\ Identity\ Manager

WINEPREFIX=$HOME/.autodesk_fusion/wineprefixes/default wine ./AdskIdentityManager.exe "THE LINK YOU COPIED FROM THE BROWSER AND PASTED IN BETWEEN THESE QUOTES"
```

**Step 5:**
download:
https://github.com/cryinkfly/Autodesk-Fusion-360-for-Linux/blob/main/files/extras/patched-dlls/Qt6WebEngineCore.dll.7z


**Step 6:**
extract to:
HOME/.fusion360/wineprefixes/default/drive_c/Program\ Files/Autodesk/webdeploy/production/<long_name_folder_with_other_dlls>

**Step 7:**
```
exit
```
```
distrobox stop fusion
```
```
distrobox enter fusion
```
```
.autodesk_fusion/bin/autodesk_fusion_launcher.sh
```

**Comment by NischalDawadi** [comment]
Following your steps it shows qt6error and fusion doesn't open at all.

**Comment by MatiM72737** [comment]
Has anyone made this work currently?

**Comment by Dav1d-Fn** [comment]
I really wish, this was working for me :/ I have a setup with an amd gpu, so i used this line without nvidia:

sudo pacman -Syu eza fzf fastfetch base-devel mokutil 

As i understood, amd drivers are automatically installed? Sry if this is a dumb question, im new to the Linux world, but i really want to use Fusion.

So what acutally happens is, that it opens after installation, but everything is black in the Fusion Window. So i cant see any Login buton at all. Any Ideas, what could be the reason? And should i replace the qt dll before login?

Im not sure if this helps, but when the Black Screen is opened, those messages come periodically in the Console:

0738:fixme:bcrypt:BCryptGenRandom ignoring selected algorithm
0738:fixme:bcrypt:BCryptGenRandom ignoring selected algorithm
0738:fixme:bcrypt:BCryptGenRandom ignoring selected algorithm
0738:fixme:bcrypt:BCryptGenRandom ignoring selected algorithm
0738:fixme:bcrypt:BCryptGenRandom ignoring selected algorithm
09dc:fixme:secur32:get_enabled_protocols handle TLS parameters
09dc:fixme:crypt:CertGetNameStringW Unsupported flags 0x10002.
09dc:fixme:crypt:CertGetNameStringW Unsupported flags 0x10002.

Setup:

Fedora 43, KDE Plasma

CPU: AMD Ryzen 7 5700X3D
GPU: AMD RX 7800 XT
RAM: 4x8 GB DDR4 Corsair (3200 MHz)
Motherboard: Asus Aorus Elite X570 (AM4)
